spec/features/posts_spec.rb

```diff
@@ -0,0 +1,57 @@
+require 'rails_helper'
+
+RSpec.feature 'Posts', type: :feature do
+  before do
+    @user = create(:user)
+    @post = create(:post, author_id: @user.id)
+  end
+
+  describe 'Index Page' do
+    it 'displays posts and user information' do
+      # Navigate to the index page
+      visit user_posts_path(user_id: @user.id)
+
+      # check for user profile picture
+      expect(page).to have_selector('img[src="https://picsum.photos/200"]')
+      # check for user name
+      expect(page).to have_content(@user.name)
+      # check for number of posts
+      expect(page).to have_content('Number of posts: 1')
```

Comment:
- [ ] To make it dynamic, instead of hardcoding your test, kindly use a method from rails to count the number of posts in the test. So no matter how posts the user has you will not have to change them manually 💯

Reply:
resolved! ✔️